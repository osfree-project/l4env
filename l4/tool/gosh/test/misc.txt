
 TITLE TEXT
 can _be_ multi-line

 Authors

; $Id$

; This document demonstrates GOSH (Gold Out of SHit) and all its implemented
; features. It will be enhanced when GOSH is enhanced <--- no advice - simple
; rule!
; BTW: Comments start with ';'.

; FIXMEs / Questions
;
; * Do NOT strip all newlines - there could be intention to use 'em.
; * What is 'EMPTY'?
; * Will "LaTeX" be factored out of GOSH like HTML?

\tableofcontents

Intro
#####

Howto build this file:


!# cat gosh.txt > out.txt
!# sed s/^/\!/ gosh.txt >> out.txt
!# gosh out.txt > gosh.tex
!# rm out.txt


*Some hidden chapter
####################

1st Order Heading -- Chapter
############################

2nd Order Haeding -- Section
============================

3rd Order Heading -- SubSection
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

4th Order Heading -- Paragraph
------------------------------


Text styles
###########

Plain text looks like this.

Paragraph 1 is here.

; comments between paragraphs are not a problem 

Paragraph 2 should be here.

Use 'apostrophes for monospaced text', but #hash# works too for compatibility.

Trigger bug in newline handling:

BUG appears if a new line starts with non-words.

BUG this should be a new paragraph, but it is'nt.
(because of the bracket)

_Just_emphasize_this_point_ but only this point.

Trigger bug in emph: (_BUG_) - *GOTCHA* This bug is no more.
What about this one: (_BUG0_,_BUG1_)?

* itemize is nice.
* what do you think?
  * Yes?
  * No?
  * Maybe!

# Let's enumerate some points.
# Test point


:Description: Use this "for descriptions."
  The "description" continues here...

  and here.

:Advice: Watch the step!


Trigger bug in itemize newline handling:

BUG paragraph one

BUG paragraph two

BUG paragraph three
* item 1
* item 2
BUG paragraph four


!Whole line is verbatim. Maybe used for sources ...
!/** MAIN */
!void main(void) {
!    exit(0);
!}


"Abstract"
##########

_TODO_


References
##########

Ref section A
=============

Forward refer to section [Ref section B].


Ref section B
=============

Refer to section [Ref section A] and trigger _linebreak_bug_ [Ref
section A].

_TODO:_ bibs, figures, ...


Tables
######

Some Text could be here. Maybe more ...


 col1 | col2 | col3
 ------------------
 col1 | col2 | col3
 col1 | col2 | col3
 col1 | col2 | col3
 col1 | col2 | col3
 ------------------
 ------------------
 col1 | col2 | col3
[table LABEL] TEXT

 col1 | col2 | col3
 ------------------
 col1 | col2 | col3
[table TABLE2] second table

 accentuation  | test
----------------------------
 italic        | _italic_
 bold          | *bold*
 monospaced    | 'monospaced'

[table accentuations full-span]

Please not that you need a _space_ in the line between table [LABEL] and 
[TABLE2] in the source representation. Table [accentuations] shows
italic, bold, and monospaced styles.
;Is it correct that this text is indented?

More Text for Tab. [LABEL] follows here.


Images
######

_TODO_

;; usage as follows
;; <image_file_without_extension> will be label with some things
;; stripped of (e.g. '_')
;[image <image_file_without_extension> <scale>% <angle>°]
;caption
;more caption


"Bibliography"
##############

_TODO_


Source file of this document
############################

A damn url [http://www.hilton.de/dispatch/ViewPropertyHomePageForm?hid=11036375&ViewPropertyHomePage=&formName=ViewPropertyHomePageForm - Hilton]

Underlines in Latex citations text [xaz_123] text

There was a bug where an
'apostroph' caused the start of a new paragraph.

There was also a bug where a
[reference] caused the start of a new paragraph.

There was even a bug where a
"quotes" caused the start of a new paragraph.

Just to test, how -- and --- are handled by the backend.
BTW, what happens when they appear inside [refer --- ences]?


[http://a.link.as.section.title]
================================

Test for monospaced text within quotes
######################################

here comes "'some quoted monospaced text'" does it work?

here comes "#some quoted monospaced text#" with a hash?

here comes "_some_quoted italic text_"

here comes "*some quoted bold text*" followed by some normal text.

Text after image without empty line
###################################

'bla.png' is missing.
;[image bla.png]
This is some plain text that was wrongly assigned to the
caption of the image.

Accentuations
#############

Monospaced and 'non'-monospaced connected by a hyphen.

Accentuation of 'IPCmon''s name.

This works
##########

* _To be written by #Krishna#_
* _To be written by *Krishna* _
* _To be written by 'Krishna' _

This don't
##########

* _To be written by *Krishna*_
* _To be written by 'Krishna'_

Annotations
###########

:description:
  Here is some normal text
  | followed by an annotation
  and some more text
