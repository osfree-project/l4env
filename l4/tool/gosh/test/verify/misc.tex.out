
\documentclass[11pt,ngerman,a4paper,normalheadings,DIV14]{scrartcl}
\usepackage{times}
\usepackage[T1]{fontenc}
\usepackage[latin1]{inputenc}
\usepackage[small,bf,hang]{caption2}
\usepackage[ngerman]{babel}
\usepackage{epsfig}
\emergencystretch = 10pt
\clubpenalty = 10000
\widowpenalty = 10000
\displaywidowpenalty = 10000
\usepackage{amsmath}
\usepackage{amssymb}
\begin{document}
\title{TITLE TEXT can \emph{be} multi-line}
\author{Authors}
\maketitle


 %     -+*|[ INTRO ]|*+-

 \section{Intro} \label{Intro}

  Howto build this file:


  \begin{verbatim}
  # cat gosh.txt > out.txt
  # sed s/^/\!/ gosh.txt >> out.txt
  # gosh out.txt > gosh.tex
  # rm out.txt
  \end{verbatim}



 %     -+*|[ 1ST ORDER HEADING -- CHAPTER ]|*+-

 \section{1st Order Heading -- Chapter} \label{1st Order Heading  Chapter}


  %     -+*|[ 2ND ORDER HAEDING -- SECTION ]|*+-

  \subsection{2nd Order Haeding -- Section} \label{2nd Order Haeding  Section}


   %     -+*|[ 3RD ORDER HEADING -- SUBSECTION ]|*+-

   \subsubsection{3rd Order Heading -- SubSection} \label{3rd Order Heading  SubSection}


    %     -+*|[ 4TH ORDER HEADING -- PARAGRAPH ]|*+-

    \paragraph{4th Order Heading -- Paragraph}



 %     -+*|[ TEXT STYLES ]|*+-

 \section{Text styles} \label{Text styles}

  Plain text looks like this.

  Paragraph 1 is here.


  Paragraph 2 should be here.

  Use \texttt{apostrophes for monospaced text}, but \texttt{hash} works too for compatibility.

  Trigger bug in newline handling:

  BUG appears if a new line starts with non-words.

  BUG this should be a new paragraph, but it is'nt.
  (because of the bracket)

  \emph{Just emphasize this point} but only this point.

  Trigger bug in emph: (\emph{BUG}) - \textbf{GOTCHA} This bug is no more.


  \begin{itemize}
   \item
    itemize is nice.
   \item
    what do you think?
    \begin{itemize}
     \item
      Yes?
     \item
      No?
     \item
      Maybe!

    \end{itemize}
  \end{itemize}
  \begin{enumerate}
   \item
    Let's enumerate some points.
   \item
    Test point


  \end{enumerate}
  \begin{description}
   \item[Description]
    Use this ``for descriptions.''
    The ``description'' continues here...

    and here.

   \item[Advice]
    Watch the step!


  \end{description}
  Trigger bug in itemize newline handling:

  BUG paragraph one

  BUG paragraph two

  BUG paragraph three
  \begin{itemize}
   \item
    item 1
   \item
    item 2
  \end{itemize}
  BUG paragraph four


  \begin{verbatim}
  Whole line is verbatim. Maybe used for sources ...
  /** MAIN */
  void main(void) {
      exit(0);
  }
  \end{verbatim}



 %     -+*|[ "ABSTRACT" ]|*+-

 \section{``Abstract''} \label{Abstract}

  \emph{TODO}



 %     -+*|[ REFERENCES ]|*+-

 \section{References} \label{References}


  %     -+*|[ REF SECTION A ]|*+-

  \subsection{Ref section A} \label{Ref section A}

   Forward refer to section \ref{Ref section B}.



  %     -+*|[ REF SECTION B ]|*+-

  \subsection{Ref section B} \label{Ref section B}

   Refer to section \ref{Ref section A} and trigger \emph{linebreak bug} [Ref
   section A].

   \emph{TODO:} bibs, figures, ...



 %     -+*|[ TABLES ]|*+-

 \section{Tables} \label{Tables}

  Some Text could be here. Maybe more ...


  \begin{table}[ht!]
  {\center
   \begin{tabular}{|l|l|l|}
   \hline
  col1& col2& col3    \\
    \hline
    \hline
    col1& col2& col3    \\
    col1& col2& col3    \\
    col1& col2& col3    \\
    col1& col2& col3    \\
      \hline
      \hline
    col1& col2& col3    \\
    \hline
   \end{tabular}
   \caption{TEXT}
   \label{LABEL}
  }
  \end{table}
  More Text for Tab. \ref{LABEL} follows here.



 %     -+*|[ IMAGES ]|*+-

 \section{Images} \label{Images}

  \emph{TODO}




 %     -+*|[ "BIBLIOGRAPHY" ]|*+-

 \section{``Bibliography''} \label{Bibliography}

  \emph{TODO}



 %     -+*|[ SOURCE FILE OF THIS DOCUMENT ]|*+-

 \section{Source file of this document} \label{Source file of this document}

  A damn url \texttt{http://www.hilton.de/dispatch/ViewPropertyHomePageForm?hid=11036375\&ViewPropertyHomePage=\&formName=ViewPropertyHomePageForm}

  Underlines in Latex citations text \cite{xaz_123} text

  There was a bug where an
  \texttt{apostroph} caused the start of a new paragraph.

  There was also a bug where a
  \cite{reference} caused the start of a new paragraph.

  There was even a bug where a
  ``quotes'' caused the start of a new paragraph.

  Just to test, how -- and --- are handled by the backend.
  BTW, what happens when they appear inside \cite{refer --- ences}?



  %     -+*|[ [HTTP://A.LINK.AS.SECTION.TITLE] ]|*+-

  \subsection{\texttt{http://a.link.as.section.title}} \label{httpalinkassectiontitle}


 %     -+*|[ TEST FOR MONOSPACED TEXT WITHIN QUOTES ]|*+-

 \section{Test for monospaced text within quotes} \label{Test for monospaced text within quotes}

  here comes ``\texttt{some quoted monospaced text}'' does it work?

  here comes ``\texttt{some quoted monospaced text}'' with a hash?

  here comes ``\emph{some quoted italic text}''

  here comes ``\textbf{some quoted bold text}'' followed by some normal text.


 %     -+*|[ TEXT AFTER IMAGE WITHOUT EMPTY LINE ]|*+-

 \section{Text after image without empty line} \label{Text after image without empty line}


  \begin{figure}[tbp]
   \begin{center}
    \epsfig{file=bla.png,angle=0,width=0.8\columnwidth}
    \caption{}
    \label{blapng}
   \end{center}
  \end{figure}

  This is some plain text that was wrongly assigned to the
  caption of the image.


 %     -+*|[ ACCENTUATIONS ]|*+-

 \section{Accentuations} \label{Accentuations}

  Monospaced and \texttt{non}-monospaced connected by a hyphen.

\newpage
\bibliographystyle{plain}
\bibliography{custom,master}
\end{document}
